<a href="https://colab.research.google.com/github/MANU7686/python-letsupgrade-assignments/blob/master/pythonday2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**DAY 2: BASIC SYNTAX**

**1) BACK SLASH : showing continuation.**


> 



In [ ]:
print("hello python")

hello python


In [ ]:
print("hello
python")

SyntaxError: ignored

here we need to add a back slash to show continuation as below


In [ ]:
print("hello\
python")

hellopython


for formatted output like tab **\t**, new line **\n**, as follows

In [ ]:
print("hello\t python")
print("hello\n python")

hello	 python
hello
 python


**2) TRIPLE QUOTE: text art,doc string,sql statements**


In [ ]:
print("""╔══╗
╚╗╔╝
╔╝(¯`v´¯)
╚══`.¸.[Python]
""")

╔══╗
╚╗╔╝
╔╝(¯`v´¯)
╚══`.¸.[Python]



3)**STRING INSIDE QUOTE:**


with single quotes,

In [ ]:
print('python world')

python world


with double quotes,


In [ ]:
print("python world")

python world


In [ ]:
the trouble maker,


In [ ]:
print('python's world')

SyntaxError: ignored

in above example it treats the apostrophe in english as single quote in python,
to resolve this add a backslash to show that it belongs to english and not python as follows,


In [ ]:
print('python\'s world')

python's world


**good to go  :)  ...!**

# **MULTIPLE STATEMENTS** 

1.   List item
2.   List item




In [10]:
name="mansi"
marks=89.8765
age=21
print("name of person is %s marks are %f age is %d"%(name,marks,age))

name of person is mansi marks are 89.876500 age is 21


to give spaces or tabs you could use following form, for float the number after . depicts the number of digits after decimal point





In [13]:
name="mansi"
marks=89.8765
age=21
print("name of person is %8s marks are %8.2f age is %8d"%(name,marks,age))

name of person is    mansi marks are    89.88 age is       21


for python version 3.4 onwards you can use following form..

In [14]:
name="mansi"
marks=89.8765
age=21
print(f" the name of person is {name} the marks are {marks} and the age is{age}")

 the name of person is mansi the marks are 89.8765 and the age is21


#**VARIABLES AND ASSIGNMENT STATEMENTS IN PYTHON**

PYTHON VARIABLE: 


*   Linking Data to name.
*   according to datatype the interpreter reserves memory space.
*   variable refers to the memory location that contains data.






In [ ]:
name="mansi"
age=21


##**Assignment statement**


In [16]:
x=y=10
id(x)


10914784

In [17]:
y=10
id(y)

10914784

here 10914784 is the base address of both x and y when we delete x the link of x to the number gets deleted but as y is linked to 10 it wont get erased as soon as del y is done the number vanishes from memory and address is null.
this happens onlu in range of (-5 to 256) lets try out of this range.

In [23]:
x=257
id(x)

139860367777168

In [24]:
y=257
id(y)

139860367776912

In [25]:
x is y

False

#**Operators**

##**Arithmetic**

###Exponential operator **
###multiplication       *
###division             /
###modulus              %
###addition             +
###subtraction          -
###floor division       //



In [29]:
a=10
b=2
print(a**b)

100


hence ** is symbol of power in python.

now let us see the differnce between // and / operator, 
###note that:
####in python 2 int/int=int
**ie.10/29=0**
####in python 3 int/int=float
**ie.10/29=0.344827**


In [30]:
a=10
b=29
print(a/b)

0.3448275862068966


####comparison operator,assignment and bitwise same  as that of c.


#**Interview Question**

**Code 1**

In [31]:
9<10 and 20>m

NameError: ignored

**Code 2**

In [32]:
9>10 and 20>m

False

###**Here in code 1 it gives error but in code 2 no error why...?**
**Ans**:
in code 1: the first equation is true so control moves to 2nd equation hence gives errror but
in code 2: as the first equation itself it false it doesnt check for second equation..;)